contracts/AelinPool.sol

```diff
@@ -385,6 +443,103 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
         emit PurchasePoolToken(msg.sender, purchaseTokenAmount);
     }
 
+    /**
+     * @dev allows anyone to become a purchaser with a qualified erc721
+     * nft in the pool depending on the scenarios
+     *
+     * Scenarios:
+     * 1. each wallet holding a qualified NFT to deposit an unlimited amount of purchase tokens
+     * 2. certain amount of purchase tokens per wallet regardless of the number of qualified NFTs held
+     * 3. certain amount of Investment tokens per qualified NFT held
+     */
+
+    function purchasePoolTokensWithNft(NftArgs[] calldata nftArgs, uint256 _purchaseTokenAmount) external lock {
+        require(hasNftList, "pool does not have an NFT list");
+        require(block.timestamp < purchaseExpiry, "not in purchase window");
+
+        uint256 maxPurchaseTokenAmount;
+
+        for (uint256 i = 0; i < nftArgs.length; i++) {
+            NftArgs memory tmpNftArg = nftArgs[i];
```

Comment:
one thing I do not like about this PR is the variable names are a bit confusing. Let's aim to make them a bit clearer if possible to make it more readable. tmpNftArg just doesn't feel like a great name

Reply:
its not the end of the world though

Reply:
yea even i was not satisfied with the name, any better suggestion for NftArgs?